client/components/happychat/index.jsx

```diff
@@ -54,51 +93,93 @@ export class Happychat extends Component {
 						'is-minimizing': isMinimizing,
 					} ) }
 				>
-					<Title onCloseChat={ this.onCloseChatTitle } />
-					<Timeline />
-					<Notices />
-					<Composer />
+					<Title onCloseChat={ this.onCloseChatTitle } translate={ translate } />
+					<Timeline
+						currentUserEmail={ currentUserEmail }
+						isCurrentUser={ isCurrentUser }
+						isExternalUrl={ isExternalUrl }
+						timeline={ timeline }
+						translate={ translate }
+						twemojiUrl={ twemojiUrl }
+					/>
+					<Notices
+						chatStatus={ chatStatus }
+						connectionStatus={ connectionStatus }
+						isServerReachable={ isServerReachable }
+						translate={ translate }
+					/>
+					<Composer
+						disabled={ disabled }
+						message={ message }
+						onSendMessage={ onSendMessage }
+						onSendNotTyping={ onSendNotTyping }
+						onSendTyping={ onSendTyping }
+						onSetCurrentMessage={ onSetCurrentMessage }
+						translate={ translate }
+					/>
 				</div>
 			</div>
 		);
 	}
 }
 
 Happychat.propTypes = {
+	chatStatus: PropTypes.string,
+	connectionStatus: PropTypes.string,
+	currentUserEmail: PropTypes.string,
+	disabled: PropTypes.bool,
 	isChatOpen: PropTypes.bool,
+	isCurrentUser: PropTypes.func,
+	isExternalUrl: PropTypes.func,
 	isMinimizing: PropTypes.bool,
+	isServerReachable: PropTypes.bool,
+	message: PropTypes.string,
 	onCloseChat: PropTypes.func,
 	onMinimizeChat: PropTypes.func,
 	onMinimizedChat: PropTypes.func,
+	onSendMessage: PropTypes.func,
+	onSendNotTyping: PropTypes.func,
+	onSendTyping: PropTypes.func,
+	onSetCurrentMessage: PropTypes.func,
 	setBlurred: PropTypes.func,
 	setFocused: PropTypes.func,
+	timeline: PropTypes.array,
+	translate: PropTypes.func,
+	twemojiUrl: PropTypes.string,
+};
+
+const isCurrentUserHelper = currentUser => ( { user_id, source } ) => {
```

Comment:
I'd like to propose that we brainstorm better name for this function :)
For me, it was a bit over-generic and I had struggled a bit about why `source === 'customer'` matters if we just want to check if a `user_id` belongs to the current user.

Based on its usage, maybe something like `isMessageFromCurrentUser` would be clearer?

Reply:
Fixed in 6902e72881ca01dedd3ec88a6289b43ae19acc98

Reply:
Fixed in 6902e72